frontends/p4/fromv1.0/converters.cpp

```diff
@@ -240,6 +242,20 @@ const IR::Node* ExpressionConverter::postorder(IR::Neq *neq) {
         return new IR::BoolLiteral(neq->srcInfo, true);  // everything else is true
 }
 
+std::map<cstring, ExpressionConverter::funcType>* ExpressionConverter::cvtForType = nullptr;
+
+void ExpressionConverter::addConverter(cstring type, ExpressionConverter::funcType cvt) {
+    static std::map<cstring, ExpressionConverter::funcType> tbl;
```

Comment:
You could use `std::type_index` directly as the key type, rather than converting to a string.